File: LeetCode/Solution_136.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        res = 0
        for num in nums:
            res ^= num
        return res
    
# Time Complexity: O(n) - We traverse the list of numbers once
# Space Complexity: O(1) - We use a constant amount of space

File: LeetCode/Solution_136.java
// 136. Single Number

class Solution_136 {
    public int singleNumber(int[] nums) {
        int res = 0;
        for(int num: nums) {
            res ^= num;
        }
        return res;
    }
}

// Time Complexity: O(n)
// Space Complexity: O(1)